base/http_listener.go

```diff
@@ -27,6 +34,25 @@ func ListenAndServeHTTP(addr string, connLimit int, certFile *string, keyFile *s
 		if err != nil {
 			return err
 		}
+
+		if clientCerts != nil {
+			// TLS client certificate support:
+			clientCAData, err := ioutil.ReadFile(clientCerts.CA_cert)
+			if err != nil {
+				return err
+			}
+			config.ClientCAs = x509.NewCertPool()
+			if !config.ClientCAs.AppendCertsFromPEM(clientCAData) {
+				return ErrInvalidCertificate
+			}
+			if clientCerts.Required == nil || *clientCerts.Required {
+				config.ClientAuth = tls.RequireAndVerifyClientCert
+				Infof(KeyHTTP, "Requiring HTTP client certs, using CA from %s, on %v", clientCerts.CA_cert, SD(addr))
+			} else {
+				config.ClientAuth = tls.VerifyClientCertIfGiven
```

Comment:
I don't completely follow the distinction between RequireAndVerifyClientCert, and VerityClientCertIfGiven.  If we're setting clientCerts.Required, should we be returning an error during auth if TLS.VerifiedChains is empty?